### PR TITLE
Add HPAv2 tutorial

### PR DESCRIPTION
**Pull request by alexellis**
Fixes: https://github.com/openfaas/faas-netes/issues/483

Signed-off-by: Alex Ellis (OpenFaaS Ltd) <alexellis2@gmail.com>

## Description

Add HPAv2 tutorial

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

Fixes: https://github.com/openfaas/faas-netes/issues/483

## How Has This Been Tested?

On an Intel NUC running Kubernetes `v1.14.1` setup via kubeadm.

Details in the tutorial of what was done

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: docs/tutorials/kubernetes-hpa.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+# watch -n 5 "kubectl describe hpa/nodeinfo -n openfaas-fn"
+```
+
+You can also monitor the replicas of your function in the OpenFaaS UI or via the CLI:
+
+```
+watch -n 5 "faas-cli list"
+```
+
+Here is ane example of the replicas scaling up in response to the traffic created by `hey`:
```
Comment (szuecs) [inline docs/tutorials/kubernetes-hpa.md]: ane->an

Comment (alexellis) [reply docs/tutorials/kubernetes-hpa.md]: Good catch. Would you mind trying this out?

**Comment by alexellis** [comment]
I think it would be useful to be able to set a memory or CPU request and limit via the CLI (without YAML). 

@rgee0 or @martindekov please could you create a feature request in openfaas/cli?

**Comment by martindekov** [comment]
I will open one in couple of hours if @rgee haven't done it already :smile:

**Comment by martindekov** [comment]
https://github.com/openfaas/faas-cli/issues/new here is the Issue 👍

File: docs/tutorials/kubernetes-hpa.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+### Disable auto-scaling with OpenFaaS
+
+Disable auto-scaling by scaling alertmanager down to zero replicas, this will stop it from firing alerts.
+
+```sh
+kubectl scale -n openfaas deploy/alertmanager --replicas=0
+```
+
+### Configure Prometheus to scrape the Kubernetes API server
```
Comment (stefanprodan) [inline docs/tutorials/kubernetes-hpa.md]: This not needed, metrics server collects it's own metrics.

File: docs/tutorials/kubernetes-hpa.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+faas-idler-67f9dcd4fc-85rbt          1m           4Mi             
+gateway-7c687d498f-nvjh8             4m           21Mi            
+nats-d4c9d8d95-fxrc2                 1m           6Mi             
+prometheus-549c7d687d-zznmq          9m           41Mi            
+queue-worker-544bcb7c67-72l72        1m           3Mi
+```
+
+> Note: you can view function usage using `-n openfaas-fn`, or core-service usage with `-n openfaas`
+
+### Disable auto-scaling with OpenFaaS
```
Comment (stefanprodan) [inline docs/tutorials/kubernetes-hpa.md]: This could be done at function level with `com.openfaas.scale.factor=0`

Comment (alexellis) [reply docs/tutorials/kubernetes-hpa.md]: Yes we could turn it off at a per function basis.

File: docs/tutorials/kubernetes-hpa.md (hunk 2 above)
Comment (alexellis) [reply docs/tutorials/kubernetes-hpa.md]: Will we need it for the custom metrics + adapter later?

**Comment by alexellis** [comment]
Thank you for the feedback Stefan

File: docs/tutorials/kubernetes-hpa.md (hunk 2 above)
Comment (stefanprodan) [reply docs/tutorials/kubernetes-hpa.md]: No, the prom adapter should target the OF Prometheus, no need to install a 2nd one.

**Comment by alexellis** [comment]
If you have time to write the extension for custom metrics you can merge it into my branch or we could do it at a later time.

**Comment by alexellis** [comment]
@stefanprodan I've added your feedback for the tutorial.

**Pull request merged**